gamemodes/terrortown/gamemode/client/cl_player_ext.lua

```diff
@@ -114,6 +114,25 @@ function plymeta:AnimUpdateGesture()
     end
 end
 
+---
+-- Creates and caches a elliptic nick for a given length.
+-- @param number width The maximum width that should be used to limit the nick
+-- @param[default="DefaultBold"] string font The font ID
+-- @param[default="..."] string limitChar The limiting character(s) that might be appended to the end
+-- @param[default=1.0] number scale The UI scale factor
+-- @return string The length limited nick
+-- @realm client
+function plymeta:NickElliptic(width, font, limitChar, scale)
+    limitChar = limitChar or "..."
+
+    self.cachedEllipticNicks = self.cachedEllipticNicks or {}
+
+    self.cachedEllipticNicks[width] = self.cachedEllipticNicks[width]
```

Comment:
This might lead to inconsistent fonts in cases where some players have a cached ellipticNick for font A and others don't. If then something like the scoreboard wants to get the elliptic nick for those players with font B, some players will retain font A through this caching, correct?

i'm not exactly sure how to nicely prevent this from happening without making this more complex though

Reply:
Additionally we probably need to throw the cache away once the players scale changes

Reply:
ah, shit, this makes caching a pain. Might not be worth it to cache it then. Right now the wrapped text in the F1 menu is recreated every frame as well. Which is way more heavy than this